data-raw/simulate_longitudinal_data.R

```diff
@@ -1,320 +1,72 @@
-  dfENROLL <- AddENROLL(df = dfENROLL,
-                        dfParticipants = dfParticipants,
-                        startDate[i],
-                        endDate[i])
-
-  # Calculate AE/SAE metrics using simulated data
-  lData <- list(
-    Raw_SUBJ = dfParticipants,
-    Raw_AE = dfAEs,
-    Raw_PD = dfPD %>%
-      rename(subjid = subjectenrollmentnumber),
-    Raw_LB = dfLB,
-    Raw_STUDCOMP = dfSTUDCOMP,
-    Raw_SDRGCOMP = dfSDRGCOMP,
-    Raw_DATACHG = dfDATACHG,
-    Raw_DATAENT = dfDATAENT %>%
-      rename(subject_nsv = subjectname),
-    Raw_QUERY = dfQUERY %>%
-      rename(subject_nsv = subjectname),
-    Raw_ENROLL = dfENROLL,
-    Raw_SITE = dfSite %>%
-      rename(studyid = protocol) %>%
-      rename(invid = pi_number) %>%
-      rename(InvestigatorFirstName = pi_first_name) %>%
-      rename(InvestigatorLastName = pi_last_name) %>%
-      rename(City = city) %>%
-      rename(State = state) %>%
-      rename(Country = country),
-    Raw_STUDY = dfStudy %>%
-      rename(studyid = StudyID) %>%
-      rename(Status = status)
-  )
-
-  mapping_wf <- MakeWorkflowList(strPath = "workflow/1_mappings")
-  mapped <- RunWorkflows(mapping_wf, lData, bKeepInputData=TRUE)
-
-  # Step 2 - Create Analysis Data - Generate 12 KRIs
-  kri_wf <- MakeWorkflowList(strPath = "workflow/2_metrics", strNames = "kri")
-  kris <- RunWorkflows(kri_wf, mapped)
-
-  cou_wf <- MakeWorkflowList(strPath = "workflow/2_metrics", strNames = "cou")
-  cous <- RunWorkflows(cou_wf, mapped)
-
-  # Step 3 - Create Reporting Data - Import Metadata and stack KRI Results
-  lReporting_Input_site <- list(
-    Raw_ctms_site = clindata::ctms_site,
-    Raw_ctms_study = clindata::ctms_study,
-    Mapped_ENROLL = mapped$Mapped_ENROLL,
-    Mapped_STUDY = mapped$Mapped_STUDY,
-    Mapped_SITE = mapped$Mapped_SITE,
-    Mapped_COUNTRY = mapped$Mapped_COUNTRY,
-    lWorkflows = kri_wf,
-    lAnalyzed = kris,
-    dSnapshotDate = endDate[i],
-    strStudyID = "AA-AA-000-0000"
-  )
-  reporting_wf_site <- MakeWorkflowList(strPath = "workflow/3_reporting")
-  reporting_site <- RunWorkflows(reporting_wf_site, lReporting_Input_site)
-
-  lReporting_Input_country <- list(
-    Raw_ctms_site = clindata::ctms_site,
-    Raw_ctms_study = clindata::ctms_study,
-    Mapped_ENROLL = mapped$Mapped_ENROLL,
-    Mapped_STUDY = mapped$Mapped_STUDY,
-    Mapped_SITE = mapped$Mapped_SITE,
-    Mapped_COUNTRY = mapped$Mapped_COUNTRY,
-    lWorkflows = cou_wf,
-    lAnalyzed = cous,
-    dSnapshotDate = endDate[i],
-    strStudyID = "AA-AA-000-0000"
-  )
-  reporting_wf_country <- MakeWorkflowList(strPath = "workflow/3_reporting")
-  reporting_country <- RunWorkflows(reporting_wf_country, lReporting_Input_country)
-
-  if (i == 1) {
-    lAnalysis_site <- kris
-    lAnalysis_country <- cous
-    lReporting_site <- reporting_site
-    lReporting_country <- reporting_country
-  }
-  else {
-  lAnalysis_site <- imap(lAnalysis_site, \(this, metric)
```

Comment:
This loop/imap step is necessary to iterate over each snapshot and then bind the reportingxxx objects together to render over time plots